numba/tests/test_analysis.py

```diff
@@ -1026,3 +1026,21 @@ def impl(a, b):
         args = (np.zeros((5, 4, 3, 2)), np.zeros((1, 1)))
 
         self.assertPreciseEqual(impl(*args), impl.py_func(*args))
+
+    def test_empty_tuple_prune(self):
```

Comment:
This test doesn't seem to cover the use of `typed_getitem`? There's no `literal_unroll`, which is the only public method of generating such an `Expr`?

Is perhaps this test instead for ensuring `isinstance`, semantic const rewrites and branch pruning can all cooperate, in which case I think a test for the `typed_getitem` case needs writing?